src/com/googlecode/jmxtrans/model/naming/KeyUtils.java

```diff
@@ -160,7 +160,7 @@ public static String getConcatedTypeNameValues(List<String> typeNames, String ty
 		StringBuilder sb = new StringBuilder();
 		for (String key : typeNames) {
 			String result = typeNameValueMap.get(key);
-			if (result != null) {
+			if (result != null && !result.isEmpty()) {
```

Comment:
That could be replaced by `Strings.isNullOrEmpty()`, which I find more readable. Or the equivalent from apache common-lang